spec/factories/products.rb

```diff
@@ -66,5 +66,17 @@
         end
       end
     end
+
+    trait :with_tags do
+        sequence(:name) { |n| "Sample Product with Tags #{n}" }
+        after(:create) do |product|
+        create_list(:category, 11).each do |category|
+          CategoryPractice.create(
+            innovable: product,
+            category: category
+          )
+        end
```

Comment:
with the suggested change to the `CategoryPractice` factory above, this can be updated to:

```
    create_list(:category, 11).each do |category|
      create(:category_practice, :for_product, innovable: product, category: category)
    end
```